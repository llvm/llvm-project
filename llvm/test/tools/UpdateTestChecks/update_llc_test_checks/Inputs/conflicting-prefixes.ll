; RUN: sed 's/RETVAL/1/g' %s | llc -mtriple=riscv32 \
; RUN:   | FileCheck -check-prefixes=CHECK,CHECKA %s
; RUN: sed 's/RETVAL/2/g' %s | llc -mtriple=riscv32 \
; RUN:   | FileCheck -check-prefixes=CHECK,CHECKA %s
; RUN: sed 's/RETVAL/3/g' %s | llc -mtriple=riscv32 \
; RUN:   | FileCheck -check-prefixes=CHECK,CHECKB %s
; RUN: sed 's/RETVAL/4/g' %s | llc -mtriple=riscv32 \
; RUN:   | FileCheck -check-prefixes=CHECK,CHECKB %s

define i32 @foo() {
  ret i32 RETVAL
}

define i32 @bar() {
  ret i32 100
}
