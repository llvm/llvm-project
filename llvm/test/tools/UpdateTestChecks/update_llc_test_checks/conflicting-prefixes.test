# REQUIRES: riscv-registered-target

# RUN: cp -f %S/Inputs/conflicting-prefixes.ll %t.ll
# RUN: %update_llc_test_checks --no-generate-body-for-unused-prefixes %t.ll 2>&1 | FileCheck %s
# RUN: diff -u %S/Inputs/conflicting-prefixes.ll.expected %t.ll

# CHECK: WARNING: For function 'foo', the following RUN lines will not generate checks due to conflicting output: RUN #1 (prefixes: CHECK, CHECKA), RUN #2 (prefixes: CHECK, CHECKA), RUN #3 (prefixes: CHECK, CHECKB), RUN #4 (prefixes: CHECK, CHECKB):
