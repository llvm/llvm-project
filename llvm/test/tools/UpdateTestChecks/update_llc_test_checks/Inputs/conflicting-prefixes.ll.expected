; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --no-generate-body-for-unused-prefixes
; RUN: sed 's/RETVAL/1/g' %s | llc -mtriple=riscv32 \
; RUN:   | FileCheck -check-prefixes=CHECK,CHECKA %s
; RUN: sed 's/RETVAL/2/g' %s | llc -mtriple=riscv32 \
; RUN:   | FileCheck -check-prefixes=CHECK,CHECKA %s
; RUN: sed 's/RETVAL/3/g' %s | llc -mtriple=riscv32 \
; RUN:   | FileCheck -check-prefixes=CHECK,CHECKB %s
; RUN: sed 's/RETVAL/4/g' %s | llc -mtriple=riscv32 \
; RUN:   | FileCheck -check-prefixes=CHECK,CHECKB %s

define i32 @foo() {
  ret i32 RETVAL
}

define i32 @bar() {
; CHECK-LABEL: bar:
; CHECK:       # %bb.0:
; CHECK-NEXT:    li a0, 100
; CHECK-NEXT:    ret
  ret i32 100
}
